src/main/java/clueless/Game.java

```diff
@@ -383,15 +384,38 @@ public void loseGame(Player loser) {
 	 * @return
 	 */
 	public Player whoHasClue(List<Card> suggestion) {
-		Player pl;
+		Player pl = null;
 		boolean playerHasClue = false;
-
+		String getPlayerChar = this.getCurrentPlayer().characterName;
+		int currIndex = getCharacterIndexInTurnOrder(getPlayerChar);		
+		int nextPlayerindex = currIndex + 1;
 		int i = 0;
-		//		while ((!playerHasClue) && i < this.characterMap.size()) {
-		//			
-		//		}
-
-		return null;
+		
+			if (nextPlayerindex >= CHARACTER_TURN_ORDER.length) 
+				{ // repoint index if currIndex is last character
+					nextPlayerindex = 0;
+				}
+			
+			while ((!playerHasClue) && i < this.characterMap.size())
+			{
+				
+				int count = 0;
+				while (count < CHARACTER_TURN_ORDER.length && nextPlayerindex != currIndex) {
+
+					if (this.isPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]) 
+							&& this.getPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]).hasClue(suggestion)) {
+						pl = this.getPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]);
```

Comment:
```suggestion
						return this.getPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]);
```

you can just return the player that has a clue here